jsk_pcl_ros/include/jsk_pcl_ros/moving_least_square_smoothing.h

```diff
@@ -0,0 +1,84 @@
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2015, JSK Lab
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/o2r other materials provided
+ *     with the distribution.
+ *   * Neither the name of the JSK Lab nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+#ifndef JSK_PCL_ROS_MOVING_LEAST_SQUARE_SMOOTHING_H_
+#define JSK_PCL_ROS_MOVING_LEAST_SQUARE_SMOOTHING_H_
+
+// ros
+#include <ros/ros.h>
+#include <ros/names.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <tf/transform_broadcaster.h>
+#include <jsk_pcl_ros/MovingLeastSquareSmoothingConfig.h>
+
+// pcl
+#include <pcl_ros/pcl_nodelet.h>
+#include <pcl/point_types.h>
+#include <pcl/common/centroid.h>
+#include <pcl/surface/mls.h>
+#include <dynamic_reconfigure/server.h>
+#include <pcl/filters/filter.h>
+#include <jsk_topic_tools/diagnostic_nodelet.h>
+
+namespace jsk_pcl_ros
+{
+  class MovingLeastSquareSmoothing: public jsk_topic_tools::DiagnosticNodelet
+  {
+  public:
+    typedef MovingLeastSquareSmoothingConfig Config;
+    MovingLeastSquareSmoothing(): DiagnosticNodelet("MovingLesatSquareSmoothing"){};
+  protected:
+    virtual void onInit();
+    virtual void subscribe();
+    virtual void unsubscribe();
+    virtual void smooth(const sensor_msgs::PointCloud2ConstPtr &input);
+    virtual void config_callback(Config &config, uint32_t level);
+
+    ros::Subscriber sub_input_;
+    ros::Publisher pub_;
+    pcl::MovingLeastSquares<pcl::PointXYZRGB, pcl::PointXYZRGB> smoother_;
```

Comment:
Do we need to hold smoother_ object?
If it does not require time-series information, do not hold in nodelet.